pkg/apiserver/apiserver.go

```diff
@@ -370,7 +371,24 @@ func write(statusCode int, apiVersion string, codec runtime.Codec, object runtim
 		io.Copy(writer, out)
 		return
 	}
-	writeJSON(statusCode, codec, object, w, isPrettyPrint(req))
+	exporter, exact := getExporter(req)
+	exportJSON(statusCode, codec, object, w, isPrettyPrint(req), exporter, exact)
+}
+
+func getExporter(req *http.Request) (export.Exporter, bool) {
+	exp := req.URL.Query().Get("export")
```

Comment:
If we do this here clients can't tell what methods support exporting.  Servers outside the Kube code base also can't provide an export behavior.  the pattern we use for optional behavior like this is to support another interface on the storage object which calls to the strategy.  I would prefer to test for the interface in apiinstaller and pass it down to the get handler, and also add the parameter to the go-restful description there.